File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int Search(int[] nums, int target) {
        var left = 0;
        var right = nums.Length-1;

        while(left<=right){
            var m = left + (right-left)/2;

            if(nums[m] == target) return m;

            if(nums[left] <= nums[m]){
                if(target >= nums[left] && target < nums[m]){
                    right = m-1;
                } else left = m+1;
            } else {
                if(target <= nums[right] && target> nums[m]){
                    left = m+1;
                } else right = m-1;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
let l = 0;
        let r = nums.length - 1;

        while (l <= r) {
            const m = l + Math.floor((r - l) / 2);
            if (nums[m] > target) {
                r = m - 1;
            } else if (nums[m] < target) {
                l = m + 1;
            } else {
                return m;
            }
        }
        return -1;
    }
}
